src/dhcp/server.c

```diff
@@ -456,7 +456,7 @@ static void init_packet(GDHCPServer *dhcp_server, struct dhcp_packet *packet,
 	packet->gateway_nip = client_packet->gateway_nip;
 	packet->ciaddr = client_packet->ciaddr;
 	dhcp_add_option_uint32(packet, DHCP_SERVER_ID,
-						dhcp_server->server_nip);
+						get_be32(&dhcp_server->server_nip));
```

Comment:
Can you explain more about this change? What is your use case

How it is related with https://github.com/albfan/miraclecast/blob/master/src/dhcp/server.c#L678 ?